metaflow/cli.py

```diff
@@ -606,11 +694,124 @@ def run(obj,
     runtime.execute()
 
 
+@parameters.add_custom_parameters
+@cli.command(help='Run the initial part of the workflow (i.e., save parameters locally).')
+@common_run_options
+@click.option('--namespace',
+              'user_namespace',
+              default=None,
+              help="Change namespace from the default (your username) to "
+                   "the specified tag. Note that this option does not alter "
+                   "tags assigned to the objects produced by this run, just "
+                   "what existing objects are visible in the client API. You "
+                   "can enable the global namespace with an empty string."
+                   "--namespace=")
+@click.pass_obj
+def pre_start(obj,
+        tags=None,
+        max_workers=None,
+        max_num_splits=None,
+        max_log_size=None,
+        decospecs=None,
+        run_id_file=None,
+        user_namespace=None,
+        **kwargs):
+
+    if namespace is not None:
+        namespace(user_namespace or None)
+    before_run(obj, tags, decospecs + obj.environment.decospecs())
+
+    runtime = NativeRuntime(obj.flow,
+                            obj.graph,
+                            obj.datastore,
+                            obj.metadata,
+                            obj.environment,
+                            obj.package,
+                            obj.logger,
+                            obj.entrypoint,
+                            obj.event_logger,
+                            obj.monitor,
+                            max_workers=max_workers,
+                            max_num_splits=max_num_splits,
+                            max_log_size=max_log_size * 1024 * 1024)
+    write_latest_run_id(obj, runtime.run_id)
+    write_run_id(run_id_file, runtime.run_id)
+
+    parameters.set_parameters(obj.flow, kwargs)
+    runtime.persist_parameters()
+
+    # FORMAT: ($1)datastore_root location \t ($2)run_id \t ($3)next_step_to_run \t ($4)task_id(of next step) \t ($5)current step \t ($6)task_id(of current step)
+    print(f"{obj.datastore.datastore_root}\t{runtime.run_id}\tstart\t1\t_parameters\t0")
+    # runtime.execute()
+
+@parameters.add_custom_parameters
+@cli.command(help='Generate the KFP YAML to run the workflow on Kubeflow Pipelines.')
+@common_run_options
+@click.option('--output-path',
+              'pipeline_output_path',
+              default=DEFAULT_KFP_YAML_OUTPUT_PATH,
+              help="the output path (or filename) of the generated KFP pipeline yaml file")
+@click.option('--code-url',
+              'code_url',
+              default=DEFAULT_FLOW_CODE_URL,
+              help="the code URL of the flow to be executed on KFP")
+@click.option('--create-new-run',
+              show_default=True,
+              default=False,
+              help="specify whether to create a new run (Note: This option will not work as expected if your local environment is not configured to connect to the Zillow KFP cluster)"
```

Comment:
let's remove references to Zillow, this should be usable by the Metaflow community.
What is the purpose of `create-new-run`, it is confusing.

Reply:
I've refactored this to something much cleaner as noted in the previous discussion. Introduced the 2 commands "run-on-kfp" and "generate-kfp-yaml" as per your suggestion